src/tests/unittests/test_pgeometry.py

```diff
@@ -62,12 +62,33 @@ def test_directionMean(self):
 
 class TestPolygonGeometry(unittest.TestCase):
 
+    def assertEmptyPolygon(self, polygon):
+        self.assertTrue(len(polygon)==0)
+
     def test_polyintersect(self):
         x1 = np.array([(0, 0), (1, 1), (1, 0)])
         x2 = np.array([(1, 0), (1.5, 1.5), (.5, 0.5)])
-        x = pgeometry.polyintersect(x1, x2)
-        self.assertEqual(3, len(x))
-        self.assertEqual(0.25, np.abs(pgeometry.polyarea(x)))
+        intersection_polygon = pgeometry.polyintersect(x1, x2)
+        self.assertEqual(3, len(intersection_polygon))
+        self.assertEqual(0.25, np.abs(pgeometry.polyarea(intersection_polygon)))
+
+    def test_polyintersect_empty_intersection(self):
+        x1 = np.array([(0, 0), (1, 1), (1, 0)])
+        x2 = 100+np.array([(1, 0), (1.5, 1.5), (.5, 0.5)])
+        intersection_polygon = pgeometry.polyintersect(x1, x2)
+        self.assertEmptyPolygon(intersection_polygon)
+
+    def test_polyintersect_identical_polygons(self):
+        x1 = np.array([(0, 0), (1, 1), (1, 0)])
+        intersection_polygon = pgeometry.polyintersect(x1, x1)
+        for pt in x1:
+            self.assertIn(pt, intersection_polygon)
+
+    def test_polyintersect_degenerate_polygons(self):
+        x1 = np.array([(0, 0)])
```

Comment:
Raises an exception in Polygon3: cPolygon.Error: Invalid polygon or contour for operation
Should be catched, now the unit test fails.

Reply:
@QFer Test is fixed. It was a little bit tricky because the exception raises was not a standard exception.